fe/fe-core/src/main/java/org/apache/doris/nereids/stats/JoinEstimation.java

```diff
@@ -177,14 +181,27 @@ private static double computeSelectivityForBuildSideWhenColStatsUnknown(Statisti
             if (cond instanceof EqualTo) {
                 EqualTo equal = (EqualTo) cond;
                 if (equal.left() instanceof Slot && equal.right() instanceof Slot) {
-                    ColumnStatistic buildColStats = buildStats.findColumnStatistics(equal.left());
-                    if (buildColStats == null) {
-                        buildColStats = buildStats.findColumnStatistics(equal.right());
+                    Slot buildSideSlot = null;
+                    if (buildStats.findColumnStatistics(equal.left()) != null) {
+                        buildSideSlot = (Slot) equal.left();
+                    } else if (buildStats.findColumnStatistics(equal.right()) != null) {
+                        buildSideSlot = (Slot) equal.right();
                     }
-                    if (buildColStats != null) {
-                        double buildSel = Math.min(buildStats.getRowCount() / buildColStats.count, 1.0);
-                        buildSel = Math.max(buildSel, UNKNOWN_COL_STATS_FILTER_SEL_LOWER_BOUND);
-                        sel = Math.min(sel, buildSel);
+                    // the processing is mainly for olap table since external table rarely has column level statistics
```

Comment:
yes, i will think a todo solution to resolve external table problem

Reply:
it seems not easy to remove this part of count usage and since it is for no-stats, it can be remained for future use and the modification will be reverted.